src/cloudWatchLogs/document/logStreamDocumentProvider.ts

```diff
@@ -35,6 +40,52 @@ export class LogStreamDocumentProvider implements vscode.TextDocumentContentProv
         }
         return content ?? ''
     }
+
+    async provideDefinition(
+        document: vscode.TextDocument,
+        position: vscode.Position,
+        token: vscode.CancellationToken
+    ): Promise<vscode.Definition | vscode.LocationLink[] | undefined> {
+        const activeUri = document.uri
+        const logGroupInfo = parseCloudWatchLogsUri(activeUri).logGroupInfo
+        if (logGroupInfo.streamName) {
+            // This means we have a stream file not a log search.
+            return
+        }
+        const curLine = document.lineAt(position.line)
+        try {
+            const streamIDMap = this.registry.getStreamIdMap(activeUri)
+            if (!streamIDMap || streamIDMap.size === 0) {
+                throw new Error(`cwl: No streamIDMap found for stream with uri ${activeUri}`)
+            }
+            const streamID = streamIDMap.get(curLine.lineNumber)
+
+            if (!streamID) {
+                throw new Error(
+                    `cwl: current line number ${curLine.lineNumber} is unregistered in streamIDMap for ${activeUri}`
+                )
+            }
+            const parameters: CloudWatchLogsParameters = {
+                limit: this.registry.configuration.get('limit', 10000),
+            }
+            logGroupInfo.streamName = streamID
+            const streamUri = createURIFromArgs(logGroupInfo, parameters)
+            const initialStreamData: CloudWatchLogsData = {
+                data: [],
+                parameters: parameters,
+                logGroupInfo: logGroupInfo,
+                retrieveLogsFunction: getLogEventsFromUriComponents,
+                busy: false,
+            }
+            await this.registry.registerLog(streamUri, initialStreamData)
+            // Set the document language
+            const doc = await vscode.workspace.openTextDocument(streamUri)
+            vscode.languages.setTextDocumentLanguage(doc, 'log')
```

Comment:
do we need a `onDidClose` handler to dispose the `streamIDMap` in the registry?

Reply:
That sounds like a good idea. Just went ahead and implemented that.